src/pages/docs/creators/ray/index.md

```diff
@@ -6,24 +6,79 @@ description: Explore how to use Ray with Golem for scaleable Python code executi
+
+The beauty of the solution is that you can run the app locally, and when you are ready, you can execute the same app on the cluster without changing a single line of your code.
+
 {% partial file="ray/learn-more-about-ray.md" /%}
 
-If you have any questions, comments insights, praises, or doubts about these docs and Ray on Golem in general please don't hesitate to reach out to us either on
+## What do you need
+
+To use Ray on Golem, you need to install the Ray framework and Ray on Golem cluster manager. 
+
+Check out [Supported versions and other limitations](/docs/creators/ray/supported-versions-and-other-limitations) to make sure Ray on Golem supports your environment.
+
+When you have the software installed, you need a cluster configuration yaml file. 
+We provide an example yaml, which contains everything needed to start up a small, free cluster running on the Golem test network.
+Check out articles about Ray on Golem [cluster yaml](/docs/creators/ray/cluster-yaml) and [Ray CLI](/docs/creators/ray/ray-cli) used to manage the cluster.
+
+When your application is ready, and you need more power, you should move to the mainnet - there are more nodes, and the nodes themselves are more powerful, but you have to pay to use them.
+
+Payments within the Golem Network happen on a blockchain called [Polygon](https://polygon.technology), you need tokens named GLM (to pay the providers) and MATIC (for payment transaction fees).
+Please read more about [mainnet payments and funding your Ray on Golem](/docs/creators/javascript/guides/switching-to-mainnet) (it is called Requestor on Golem Network)
+
+When you have the tokens you need to configure `network: "polygon"` in cluster yaml ([more details](/docs/creators/ray/cluster-yaml#network))
+
+Check out [the machines](https://stats.golem.network/network/providers/online) ready to execute your payloads.
+
+## Life cycle of working with Ray on Golem
+
+The basic flow of working with Ray and Ray on Golem consists of:
+
+- Parallelizing your Python code with Ray (or writing a Ray app from scratch)
+- Testing your Ray app on your computer (Ray helps you utilize all of your machine's CPU cores)
+- Starting up a Ray on Golem cluster initially consisting of only one node
+- Executing your app on Ray on Golem cluster, which automatically scales up and down to accommodate your app's needs
+- Stopping down your Ray on Golem cluster to free resources and to limit the cost
+
+## How to start
+
+You need a piece of code to execute on Golem. Once you have a Ray app, you can immediately proceed to [launching it on Golem](/docs/creators/ray/setup-tutorial).
+You can also use [our example apps](https://github.com/golemfactory/golem-ray/tree/main/examples) to play with Ray on Golem. 
+
+You can also check out a more detailed explanation of [simple ray tasks app](/docs/creators/ray/basic-ray-tasks-usage-tutorial) and a more sophisticated [bridge simulation app](/docs/creators/ray/conversion-to-ray-on-golem-tutorial).
```

Comment:
```suggestion
You'll need a piece of code to execute on Golem. If you already have some Ray application, you can immediately proceed to [launching it on Golem](/docs/creators/ray/setup-tutorial).
Otherwise, feel free to experiment with [our example apps](https://github.com/golemfactory/golem-ray/tree/main/examples). 

Once you get the hang of it, we invite you to have a look at the detailed explanation of [simple ray tasks app](/docs/creators/ray/basic-ray-tasks-usage-tutorial) and a more sophisticated [bridge simulation app](/docs/creators/ray/conversion-to-ray-on-golem-tutorial).
```